pkg/marathon/client.go

```diff
@@ -64,5 +68,40 @@ func (c *Client) GroupsWithoutRootSlash() (map[string]bool, error) {
 	default:
 		return nil, errors.New("unable to get Marathon groups")
 	}
+}
+
+// Info returns the content of the Marathon endpoint 'v2/info'.
+func (c *Client) Info() (map[string]interface{}, error) {
+	dcosClient := pluginutil.HTTPClient(c.baseURL)
+	resp, err := dcosClient.Get("/v2/info")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case 200:
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		var result map[string]interface{}
+		err = json.Unmarshal(data, &result)
+		if err != nil {
+			return nil, err
+		}
+		return result, nil
+	default:
+		return nil, httpResponseToError(resp)
+	}
+}
 
+func httpResponseToError(resp *http.Response) error {
+	if resp.StatusCode < 400 {
+		return fmt.Errorf("unexpected status code %d", resp.StatusCode)
+	}
+	return &httpclient.HTTPError{
```

Comment:
I really don't like this error it has no more information then one above

https://github.com/dcos/dcos-core-cli/blob/55b3085ed15a1b26df8f0fbce58c15958c2541cc/vendor/github.com/dcos/dcos-cli/pkg/httpclient/client.go#L55-L57

Reply:
But this is the part of bigger discussion how to handle API errors

Reply:
In marathon I think we should stream resposne to a client. E.g. 500 error can happen when there is no leader and this will be clear to the user whats wrong

Reply:
IMO we should do like in the Mesos client (e.g. https://github.com/dcos/dcos-core-cli/blob/4f7337425d6506bbbf8f7d31b89056307e344d4d/pkg/mesos/client.go#L134), if we see specific HTTP code in the Marathon REST API defined in https://mesosphere.github.io/marathon/api-console/index.html we should handle them in the client. 

In the case of `v2/info`, only `200` is expected which explains the lack of further handling.